### DISCUSSION TITLE
Can we do “predictive text” for time series?

### DISCUSSION THREAD

**Discussion by albert1480** [TDAmeritrade/stumpy#448] [Help: Coding & Implementations]
<img width="1206" alt="Снимок экрана 2021-07-27 в 14 17 52" src="https://user-images.githubusercontent.com/60170640/127145260-ea01b49f-2046-421f-9831-a5d5894a2281.png">
Hi all! Can anybody provide a tutorial for this?

**Comment by seanlaw** [comment]
@artem1480 Instead of an image can you please provide the source for the original document and what you are attempting to accomplish?

**Reply by albert1480** [reply]
Dear Sean,
Here it is [https://www.cs.ucr.edu/~eamonn/100_Time_Series_Data_Mining_Questions__with_Answers.pdf](url)

**Reply by albert1480** [reply]
I found it here [https://stumpy.readthedocs.io/en/latest/](url) ''and more..."

**Reply by seanlaw** [reply]
Thanks. Can you provide some details on what you've already tried with STUMPY? What is your data?

**Reply by albert1480** [reply]
I work with patterns in 2 ts. And now have 2 dfs (csv) and want to predict as show. TS it is a simple signal from sensors.

**Reply by albert1480** [reply]
after all I want to input simple audio (sine) signal as ts and predict it

**Reply by albert1480** [reply]
the main work is finding pattern cross two ts

**Reply by seanlaw** [reply]
Have you looked at these two existing tutorials yet?

1. https://stumpy.readthedocs.io/en/latest/Tutorial_AB_Joins.html
2.  https://stumpy.readthedocs.io/en/latest/Tutorial_Pattern_Searching.html

**Reply by seanlaw** [reply]
According to the original work (not by us), they state that:

```
% The tool predicts the next subsequence based on the current subsequence
% If the current subsequence (S_current) is very similar to a subsequence
% in the past (S_past), then our best guess of the next subsequence is what
% followed S_past.
% The best guess is associated with a confidence level. The confidence
% level is the correlation between S_current and S_past.
% Only parameters data and sub_len are compulsory. The rest is optional.
% The code below is heavy on the visualization side.
% The core idea is given a query subsequence and a longer time series,
% compute the best matches with z-normalized Euclidean distance fast.
```

**Reply by albert1480** [reply]
ok(
